moler/device/textualdevice.py

```diff
@@ -310,9 +312,17 @@ def _set_state(self, state):
         if self.current_state != state:
             self._log(logging.INFO, "Changed state from '%s' into '%s'" % (self.current_state, state))
             self.SM.set_state(state=state)
+        if self._kept_state is not None and self.current_state != self._kept_state:
+            state = self._kept_state
+            try:
+                self.goto_state(state=state, timeout=self.timeout_keep_state, rerun=0,
+                                send_enter_after_changed_state=False, log_stacktrace_on_fail=False, keep_state=True)
+            except DeviceChangeStateFailure:
+                self._log(level=logging.WARNING, msg="Cannot properly go to state: '{}' in background.".format(state))
```

Comment:
Warning level give us too much not good looking logs in log.html report of test, change it e.g to debug level,
I suggest warn about unexpected state changed, then set counter for trying goto state and warn also about the last completed state change to kept state.

Reply:
Done.